### ISSUE TITLE
Listen for messages from server

### ISSUE THREAD

**Issue by pereslavtsev** [jely2002/nest-ipc#3]
Is there way to listen messages from server in client?

**Comment by jely2002** [comment]
Hi! Thanks for submitting this issue, the answer is about > 2 years late but here goes:

There's 2 patterns you can use that make this possible.

**1. Use both the client and server in the same app.**

```ts
@Controller()
export class AppController {
  constructor(private readonly ipcService: IpcService) {}

  @Post()
  broadcastHello(): void {
    this.ipcService.emit('ping', 'hello');
  }

  @SubscribeIpcMessage('pong')
  handleWorldMessage(@Payload() data: string) {
    console.log(data);
  }

}
```

In this case you emit a "ping" over IPC. The remote app then responds with "pong" and your data back, which you listen to by subscribing to the "pong" message.

**2. Use the .send method to wait for an answer back.**
```ts
@Controller()
export class AppController {
  constructor(private readonly ipcService: IpcService) {}

  @Get()
  getOverIpc(): Promise<string> {
    return firstValueFrom(this.ipcService.send('message', 'hello'));
  }
}
```

Here we send the "message" command with our data and then wait for a response. This would require the remote application to respond back with the same pattern "message" to give back data.

In short: depending on how the remote IPC application functions you've got 2 options.
